front/lib/api/assistant/conversation.ts

```diff
@@ -344,6 +344,10 @@ export async function getConversation(
     return new Err(new ConversationError("conversation_not_found"));
   }
 
+  if (!canAccessConversation(auth, conversation)) {
```

Comment:
This was leaving at the endpoint level, before?

Reply:
Just want to make sure that all downstream code do have proper auth object.

Reply:
Hmmm I don't think this was properly rolled out to all endpoints. Most endpoints rely on `getConversationWitoutContent` which does check for that. But here are a number of other endpoints that rely on getConversation that don't perform that check properly.

Eg, looking at pages/api/v1/w/[wId]/assistant/conversations/[cId]/index.ts not sure what prevents me from looking at a conversation I'm not supposed to have access to

Reply:
🤔 Pretty sure this was covered. Probably in the endpoint to fetch the pages of messages.

Reply:
Prove it to me :)

Reply:
There is a check on fetching the message here: https://github.com/dust-tt/dust/blob/aeceabc74a6666ee55c416b7cf212ce0b9c675a8/front/lib/api/assistant/messages.ts#L383.

Which ensures that the current `auth` object has access to the message. Which is definitely not ideal.

Reply:
Yep 🙌